kafka-connect-fitbit-source/src/main/java/org/radarbase/connect/rest/fitbit/request/FitbitRequestGenerator.java

```diff
@@ -75,35 +75,46 @@ public void initialize(RestSourceConnectorConfig config) {
     FitbitRestSourceConnectorConfig fitbitConfig = (FitbitRestSourceConnectorConfig) config;
     this.baseClient = new OkHttpClient();
 
-    this.userRepository = fitbitConfig.getUserRepository();
+    this.userRepositories = fitbitConfig.getUserRepositories();
     this.routes = getRoutes(fitbitConfig);
 
     super.initialize(config);
   }
 
   private List<RequestRoute> getRoutes(FitbitRestSourceConnectorConfig config) {
     AvroData avroData = new AvroData(20);
-    List<RequestRoute> localRoutes = new ArrayList<>(5);
-    localRoutes.add(new FitbitSleepRoute(this, userRepository, avroData));
-    localRoutes.add(new FitbitTimeZoneRoute(this, userRepository, avroData));
-    localRoutes.add(new FitbitActivityLogRoute(this, userRepository, avroData));
-    if (config.hasIntradayAccess()) {
-      localRoutes.add(new FitbitIntradayStepsRoute(this, userRepository, avroData));
-      localRoutes.add(new FitbitIntradayHeartRateRoute(this, userRepository, avroData));
+    List<RequestRoute> localRoutes = new ArrayList<>(5 * userRepositories.size());
+    for(UserRepository userRepository : userRepositories) {
+      localRoutes.add(new FitbitSleepRoute(this, userRepository, avroData));
+      localRoutes.add(new FitbitTimeZoneRoute(this, userRepository, avroData));
+      localRoutes.add(new FitbitActivityLogRoute(this, userRepository, avroData));
+      if (config.hasIntradayAccess()) {
+        localRoutes.add(new FitbitIntradayStepsRoute(this, userRepository, avroData));
+        localRoutes.add(new FitbitIntradayHeartRateRoute(this, userRepository, avroData));
+      }
     }
     return localRoutes;
   }
 
-  public OkHttpClient getClient(User user) {
-    return clients.computeIfAbsent(user.getId(), u -> baseClient.newBuilder()
-          .authenticator(new TokenAuthenticator(user, userRepository))
-          .build());
+  public OkHttpClient getClient(User user) throws IOException, NoSuchMethodException {
```

Comment:
Why would we request a client a client for a user that is not available in the repository? For every user, checking whether a user is available in the repository looks like an overhead to me, specially when you have 100s of uses. If possible, we should do this check outside this function.

Reply:
Ok optimised now. Thanks